satsense/generators.py

```diff
@@ -61,17 +92,36 @@ def load_image(self, itype, windows):
         self._padding = tuple(
             max(math.ceil(0.5 * w[i]) for w in windows) for i in range(2))
 
+        block = self.get_blocks()
+        image = self.image.copy_block(block)
+        self._image_cache = image[itype]
+        self.loaded_itype = itype
+
+    def get_blocks(self):
         block = []
         for i in range(2):
-            offset = math.floor((self.offset[i] + 0.5) * self.step_size[i])
+            offset = self.offset[i] * self.step_size[i]
             start = offset - self._padding[i]
-            end = (
-                offset + self._padding[i] + self.shape[i] * self.step_size[i])
+            end = (offset + self._padding[i] +
+                   (self.shape[i] * self.step_size[i]))
             block.append((start, end))
-        block = tuple(block)
-        image = self.image.copy_block(block)
-        self._image_cache = image[itype]
-        self.loaded_itype = itype
+
+        return tuple(block)
+
+    def get_slices(self, index, window):
```

Comment:
```suggestion
    def _get_slices(self, index, window):
```
This should probably be a private method